internal/operators/manager.go

```diff
@@ -0,0 +1,163 @@
+}
+
+// NewManagerWithConfig creates new instance of an Operator Manager
+func NewManagerWithConfig(log logrus.FieldLogger, hostApi host.API, cfg *ocs.Config) Manager {
+	ocsValidator := ocs.NewOcsValidator(log.WithField("pkg", "ocs-operator-state"), hostApi, cfg)
+	return Manager{
+		log:                log,
+		ocsValidatorConfig: cfg,
+		ocsValidator:       ocsValidator,
+	}
+}
+
+// GenerateManifests generates manifests for all enabled operators.
+// Returns map assigning manifest content to its desired file name
+func (mgr *Manager) GenerateManifests(cluster *common.Cluster) (map[string]string, error) {
+	lsoEnabled := false
+	ocsEnabled := mgr.checkOcsEnabled(cluster)
+	if ocsEnabled {
+		lsoEnabled = true // if OCS is enabled, LSO must be enabled by default
+	} else {
+		lsoEnabled = mgr.checkLsoEnabled(cluster)
+	}
+	operatorManifests := make(map[string]string)
+
+	if lsoEnabled {
+		manifests, err := mgr.generateLsoManifests(cluster)
+		if err != nil {
+			mgr.log.Error("Cannot generate LSO manifests due to ", err)
+			return nil, err
+		}
+		for k, v := range manifests {
+			operatorManifests[k] = v
+		}
+	}
+
+	if ocsEnabled {
+		manifests, err := mgr.generateOcsManifests(cluster)
+		if err != nil {
+			mgr.log.Error("Cannot generate OCS manifests due to ", err)
+			return nil, err
+		}
+		for k, v := range manifests {
+			operatorManifests[k] = v
+		}
+	}
+	return operatorManifests, nil
+}
+
+// AnyOperatorEnabled checks whether any operator has been enabled for the given cluster
+func (mgr *Manager) AnyOperatorEnabled(cluster *common.Cluster) bool {
+	return mgr.checkLsoEnabled(cluster) || mgr.checkOcsEnabled(cluster)
+}
+
+// ValidateOCSRequirements validates OCS requirements. Returns "true" if OCS operator is not deployed
+func (mgr *Manager) ValidateOCSRequirements(cluster *common.Cluster) bool {
+	if isEnabled(cluster, models.OperatorTypeOcs) {
+		return mgr.ocsValidator.ValidateOCSRequirements(&cluster.Cluster)
+	}
+	return true
+}
+
+// GetOperatorStatus gets status of an operator of given type.
+func (mgr *Manager) GetOperatorStatus(cluster *common.Cluster, operatorType models.OperatorType) string {
+	operator, err := findOperator(cluster, operatorType)
+	if err != nil {
+		return "Something went wrong with Unmarshalling operators"
+	}
+	if operator != nil {
+		if swag.BoolValue(operator.Enabled) {
+			return operator.Status
+		}
+	}
+	return "OCS is disabled"
+}
+
+func (mgr *Manager) generateLsoManifests(cluster *common.Cluster) (map[string]string, error) {
+	mgr.log.Info("Creating LSO Manifests")
+	return lso.Manifests(cluster.OpenshiftVersion)
+}
+
+func (mgr *Manager) generateOcsManifests(cluster *common.Cluster) (map[string]string, error) {
+	mgr.log.Info("Creating OCS Manifests")
+	return ocs.Manifests(mgr.ocsValidatorConfig.OCSMinimalDeployment, cluster.OpenshiftVersion, mgr.ocsValidatorConfig.OCSDisksAvailable, len(cluster.Cluster.Hosts))
+}
+
+func (mgr *Manager) checkLsoEnabled(cluster *common.Cluster) bool {
```

Comment:
Can we take this opportunity to apply the [golang coding conventions](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) for acronyms?
```
Words in names that are initialisms or acronyms (e.g. "URL" or "NATO") have a consistent case. For example, "URL" should appear as "URL" or "url" (as in "urlPony", or "URLPony"), never as "Url".
```

Reply:
:+1: